Unable to build lifecycle in macos darwin for Linux

### Summary
Unable to build lifecycle image for Linux due to mismatch in comparison of Docker architecture and value for GOARCH. 

Docker returns the architecture as _x86_64_ on Docker Desktop for Mac and also on my Linux VM. From what I know and from this [docker document](https://github.com/docker-library/official-images#architectures-other-than-amd64) , we should not fail the build.

Output of `docker info`
```
// on Docker Desktop on Mac OS Darwin: 
Server Version: 20.10.6
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Kernel Version: 5.10.25-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64

// on Linux VM
Server Version: 1.13.1
Runtimes: docker-runc runc
Default Runtime: docker-runc
Kernel Version: 3.10.0-1160.31.1.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.9 (Maipo)
OSType: linux
Architecture: x86_64
```


---

### Reproduction

##### Steps
<!--- What steps should be taken to reproduce the issue? -->

1. Clone the repo on a Mac OS machine
2. Run `make build-image-linux-amd64`

##### Current behavior
```
macOS❯ make build-image-linux-amd64                                                                                                                                1m 12s 10:24:12
docker build ./tools --build-arg from_image=golang:1.16-alpine --tag lifecycle-img
Sending build context to Docker daemon  23.04kB
Step 1/3 : ARG from_image
Step 2/3 : FROM ${from_image}
 ---> 7762f5dece68
Step 3/3 : RUN apk add build-base
 ---> Using cache
 ---> 40d0417fcbdf
Successfully built 40d0417fcbdf
Successfully tagged lifecycle-img:latest

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
> Building lifecycle/lifecycle for linux/amd64...
mkdir -p /lifecycle/out/linux-amd64/lifecycle
docker run \
	  --workdir=/lifecycle \
	  --volume /lifecycle/out/linux-amd64/lifecycle:/out \
	  --volume /lifecycle:/lifecycle \
	  --volume gocache:/go \
	  lifecycle-img \
	  sh -c 'GOARCH=amd64 CGO_ENABLED=0 go build  -ldflags "-s -w -X 'github.com/buildpacks/lifecycle/cmd.SCMRepository=github.com/buildpacks/lifecycle' -X 'github.com/buildpacks/lifecycle/cmd.SCMCommit=d1749ed' -X 'github.com/buildpacks/lifecycle/cmd.Version=0.11.2-166+d1749ed'" -o /out/lifecycle -a ./cmd/lifecycle'
> Creating phase symlinks for linux/amd64...
ln -sf lifecycle /lifecycle/out/linux-amd64/lifecycle/detector
ln -sf lifecycle /lifecycle/out/linux-amd64/lifecycle/analyzer
ln -sf lifecycle /lifecycle/out/linux-amd64/lifecycle/restorer
ln -sf lifecycle /lifecycle/out/linux-amd64/lifecycle/builder
ln -sf lifecycle /lifecycle/out/linux-amd64/lifecycle/exporter
ln -sf lifecycle /lifecycle/out/linux-amd64/lifecycle/rebaser
ln -sf lifecycle /lifecycle/out/linux-amd64/lifecycle/creator
> Building lifecycle/launcher for linux/amd64...
mkdir -p /lifecycle/out/linux-amd64/lifecycle
GOARCH=amd64 CGO_ENABLED=0 go build  -ldflags "-s -w -X 'github.com/buildpacks/lifecycle/cmd.SCMRepository=github.com/buildpacks/lifecycle' -X 'github.com/buildpacks/lifecycle/cmd.SCMCommit=d1749ed' -X 'github.com/buildpacks/lifecycle/cmd.Version=0.11.2-166+d1749ed'" -o /lifecycle/out/linux-amd64/lifecycle/launcher -a ./cmd/launcher
test $(du -m /lifecycle/out/linux-amd64/lifecycle/launcher|cut -f 1) -le 3
> Packaging lifecycle for linux/amd64...
go run ./tools/packager/main.go --inputDir /lifecycle/out/linux-amd64/lifecycle -archivePath /lifecycle/out/lifecycle-v0.11.2-166+d1749ed+linux.x86-64.tgz -descriptorPath lifecycle.toml -version 0.11.2-166+d1749ed
go run ./tools/image/main.go -daemon -lifecyclePath /lifecycle/out/lifecycle-v0.11.2-166+d1749ed+linux.x86-64.tgz -os linux -arch amd64 -tag lifecycle:d1749ed
2021/07/21 12:11:36 Target architecture[amd64] and daemon architecture[x86_64] must match.
exit status 1
make: *** [build-image-linux-amd64] Error 1
```

##### Expected
Build should be successful


---

### Context

##### lifecycle version
<!--- If you can find this, it helps us pin down the issue. For example, run `pack inspect-builder BUILDER` which should report the lifecycle version in question. -->

##### platform version(s)
<!--- For example run `pack report` and `docker info` and copy output here. -->

##### anything else?

--- a/tools/image/main.go
+++ b/tools/image/main.go
@@ -82,7 +82,11 @@ func main() {
 		if info.OSType != targetOS {
 			log.Fatal("Target OS and daemon OS must match")
 		}
-		if info.Architecture != targetArch {
+		daemonArch := info.Architecture
+		if daemonArch == "x86_64" {
+			daemonArch = "amd64"
+		}
+		if daemonArch != targetArch {
 			log.Fatal("Target architecture and daemon architecture must match")
 		}
 		err = pullImage(dockerClient, baseImage)

